### PR TITLE
Move common pipeline variables into a template

### PR DESCRIPTION
**Pull request by macborow**
Attempt to extract pipeline variables into a shared template file.

### CODE REVIEW

**Comment by macborow** [comment]
Variable templates did not work for our use case at the time I added the original build configuration files. It is now possible to share a single template file with common build settings like version information, so this seems like a nice clean up. I'll leave it up to you to decide whether to merge it.
Please review the modifications in case I made a copy paste mistake somewhere, I wasn't able to test this for all pipelines (PR validation worked, so it seems like safe change).

File: pipelines/config/settings.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+variables:
+  Unity2018Version: Unity2018.4.6f1
```
Comment (keveleigh) [inline pipelines/config/settings.yml]: There are slightly different Unity versions on the internal build machines as the externally-facing ones. I'm not sure the internal machines have this build, and were previously using `.12` (while external used this build)

Comment (macborow) [reply pipelines/config/settings.yml]: Ah, that's an interesting point. I blamed any variation on copy-paste mistakes as there is a lot of duplication. I would expect that the version should not differ outside of the "internal/external agent pool" split, so we could perhaps have twi sets of those variables (e.g. Unity2018Version and Unity2018VersionAnalog, etc.). I'll see if I can update the config to always pass the version down in template parameters.

**Pull request merged**